src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -34,6 +35,15 @@
 from azure.cli.command_modules.network.zone_file.parse_zone_file import parse_zone_file
 from azure.cli.command_modules.network.zone_file.make_zone_file import make_zone_file
 
+logger = azlogging.get_az_logger(__name__)
+
+def _upsert(collection, obj, key_name, key_value):
+    match = next((x for x in collection if getattr(x, key_name) == key_value), None)
+    if match:
+        logger.warning("Item '%s' already exists. Replacing with new values.", key_value)
```

Comment:
How harmful it is for a value to be replaced here? Since the method name is `upsert` I assume the update action should be allowed, so what's the value of log a warning here?